rebaser.go

```diff
@@ -50,41 +56,64 @@ func (r *Rebaser) Rebase(workingImage imgutil.Image, newBaseImage imgutil.Image,
 	// get existing metadata label
 	var origMetadata files.LayersMetadataCompat
 	if err = image.DecodeLabel(workingImage, platform.LifecycleMetadataLabel, &origMetadata); err != nil {
-		return RebaseReport{}, errors.Wrap(err, "get image metadata")
+		return RebaseReport{}, fmt.Errorf("get image metadata: %w", err)
 	}
 
 	// rebase
 	if err = workingImage.Rebase(origMetadata.RunImage.TopLayer, newBaseImage); err != nil {
-		return RebaseReport{}, errors.Wrap(err, "rebase app image")
+		return RebaseReport{}, fmt.Errorf("rebase app image: %w", err)
 	}
 
 	// update metadata label
 	origMetadata.RunImage.TopLayer, err = newBaseImage.TopLayer()
 	if err != nil {
-		return RebaseReport{}, errors.Wrap(err, "get rebase run image top layer SHA")
+		return RebaseReport{}, fmt.Errorf("get rebase run image top layer SHA: %w", err)
 	}
 	identifier, err := newBaseImage.Identifier()
 	if err != nil {
-		return RebaseReport{}, errors.Wrap(err, "get run image id or digest")
+		return RebaseReport{}, fmt.Errorf("get run image id or digest: %w", err)
 	}
 	origMetadata.RunImage.Reference = identifier.String()
-	if appPlatformAPI != "" && api.MustParse(appPlatformAPI).AtLeast("0.12") {
+	if appPlatformAPI != "" && r.PlatformAPI.AtLeast("0.12") {
 		// update stack and runImage if needed
-		if !origMetadata.RunImage.Contains(newBaseImage.Name()) {
-			origMetadata.RunImage.Image = newBaseImage.Name()
-			origMetadata.RunImage.Mirrors = []string{}
-			newStackMD := origMetadata.RunImage.ToStack()
-			origMetadata.Stack = &newStackMD
+		if !containsName(origMetadata, newBaseImage.Name()) {
```

Comment:
Should there be an else case here that also updates the metadata? If, for instance, there was an app image that had something like

`mybase/blah` and mirror of `mybase-mirror/blah` and now they've changed to `mybase/blah` and `mybase-mirrornew/blah`. I think we'd want to re-write the metadata to match so that the new mirror is propagated forward.

Reply:
But how would the new mirror be communicated to the rebaser? We can learn about a new run image via `-run-image` or `CNB_RUN_IMAGE`, but where would the mirror information come from?

Reply:
oh ok - I think I got confused a bit.  Ignore this 😄